src/System.Resources.Extensions/src/System/Resources/Extensions/DeserializingResourceReader.cs

```diff
@@ -0,0 +1,204 @@
+                        int length = _store.Read7BitEncodedInt();
+                        if (length < 0)
+                        {
+                            throw new BadImageFormatException(SR.Format(SR.BadImageFormat_ResourceDataLengthInvalid, length));
+                        }
+
+                        byte[] data = _store.ReadBytes(length);
+
+                        TypeConverter converter = TypeDescriptor.GetConverter(type);
+
+                        if (converter == null)
+                        {
+                            throw new TypeLoadException(SR.Format(SR.TypeLoadException_CannotLoadConverter, type));
+                        }
+
+                        value = converter.ConvertFrom(data);
+                        break;
+                    }
+                case SerializationFormat.TypeConverterString:
+                    {
+                        string stringData = _store.ReadString();
+
+                        TypeConverter converter = TypeDescriptor.GetConverter(type);
+
+                        if (converter == null)
+                        {
+                            throw new TypeLoadException(SR.Format(SR.TypeLoadException_CannotLoadConverter, type));
+                        }
+
+                        value = converter.ConvertFromInvariantString(stringData);
+                        break;
+                    }
+                case SerializationFormat.ActivatorStream:
+                    {
+                        // read length
+                        int length = _store.Read7BitEncodedInt();
+                        if (length < 0)
+                        {
+                            throw new BadImageFormatException(SR.Format(SR.BadImageFormat_ResourceDataLengthInvalid, length));
+                        }
+                        Stream stream;
+
+                        if (_store.BaseStream is UnmanagedMemoryStream ums)
+                        {
+                            // For the case that we've memory mapped in the .resources
+                            // file, just return a Stream pointing to that block of memory.
+                            unsafe
+                            {
+                                stream = new UnmanagedMemoryStream(ums.PositionPointer, length, length, FileAccess.Read);
+                            }
+                        }
+                        else
+                        {
+
+                            byte[] bytes = _store.ReadBytes(length);
+                            // Lifetime of memory == lifetime of this stream.
+                            stream = new MemoryStream(bytes, false);
+                        }
+
+                        value = Activator.CreateInstance(type, new object[] { stream });
+                        break;
+                    }
+                default:
+                    throw new BadImageFormatException(SR.BadImageFormat_TypeMismatch);
+            }
+
+            if (value.GetType() != type)
```

Comment:
>if (value.GetType() != type) [](start = 12, length = 28)

is it even possible?

Reply:
Yes it is possible.  For example a typeconverter doesn't return the type it is supposed to, or the binary formatter blob doesn't match the type that was declared.  A similar check is present in ResourceReader.  I can add a comment.

Reply:
Comment will be good. Thanks!